src/OpenLoco/src/GameSaveCompare.cpp

```diff
@@ -0,0 +1,503 @@
+                        gameState1.routings[route][routePerVehicle],
+                        gameState2.routings[route][routePerVehicle]);
+                }
+            }
+        }
+        if (!displayAllDivergences && divergentBytesTotal > 0)
+        {
+            Logging::info(" {} other diverging bytes omitted", divergentBytesTotal);
+        }
+        return divergentBytesTotal > 0;
+    }
+
+    bool compareElements(const std::vector<S5::TileElement>& tileElements1, const std::vector<S5::TileElement>& tileElements2, bool displayAllDivergences)
+    {
+        long divergentBytesTotal = 0;
+        std::vector<S5::TileElement> smaller = tileElements1;
+        std::vector<S5::TileElement> larger = tileElements2;
+        if (tileElements1.size() > tileElements2.size())
+        {
+            smaller = tileElements2;
+            larger = tileElements1;
+        }
+
+        if (tileElements1.size() != tileElements2.size())
+        {
+            Logging::info("The TileElements sizes are different. Will compare up to the smallest TileElements size.");
+            Logging::info("Size of TileElements1 = {}", tileElements1.size());
+            Logging::info("Size of TileElements2 = {}", tileElements2.size());
+        }
+        int elementCount = 0;
+        for (auto iterator1 = smaller.begin(), iterator2 = larger.begin(); iterator1 != smaller.end();
+             ++iterator1, ++iterator2)
+        {
+            if (iterator2 != larger.end())
+            {
+                auto tile1 = tileElements1.at(elementCount);
+                auto tile2 = tileElements2.at(elementCount);
+
+                if (!bitWiseEqual(tile1, tile2))
+                {
+                    if (divergentBytesTotal == 0)
+                    {
+                        Logging::info("DIVERGENCE");
+                        Logging::info("TILE ELEMENT[{}]", elementCount);
+                    }
+                    divergentBytesTotal = bitWiseLogDivergence("Elements[" + std::to_string(elementCount) + "]", tile1, tile2, displayAllDivergences, divergentBytesTotal);
+                }
+                elementCount++;
+            }
+        }
+        if (!displayAllDivergences && divergentBytesTotal > 0)
+        {
+            Logging::info(" {} other diverging bytes omitted", divergentBytesTotal);
+        }
+        return not(divergentBytesTotal > 0);
+    }
+
+    bool compareGameStates(const fs::path& path)
+    {
+        Logging::info("Comparing reference file {} to current GameState frame", path);
+        return compareGameStates(getGameState(), S5::importSave(path).get()->gameState, false);
+    }
+
+    bool compareGameStates(const fs::path& path1, const fs::path& path2, bool displayAllDivergences)
+    {
+        Logging::info("Comparing game state files:");
+        Logging::info("   file1: {}", path1);
+        Logging::info("   file2: {}", path2);
+
+        S5::setGameState(S5::importSave(path1));
```

Comment:
There is no need to set this just import the state to one variable and the other state to another.

```
auto state1 = S5::importSave(path1);
auto state2 = S5::importSave(path2);
auto match = compareGameStates(state1->gameState, state2->gameState, displayAllDivergences);
match |= compareElements(state1->tileElements, state2->tileElements, displayAllDivergences);
```